### PR TITLE
Convert filter buttons into a ButtonGroup to toggle tables in the Model Details view...and more...

### PR DESCRIPTION
**Pull request by hatched**
## Done

- Converted `ModelGroupToggle` component to a generic `ButtonGroup` component.
- Fixed bug in `FilterTags` that caused multiple history entries to be added if multiple components tried to update the query strings.
- Fixed bug in `LogIn` that caused the whole application to be destroyed and re-rendered on initial load.
- Created `useQueryString` hook that can be used directly by the `ButonGroup` component, or independently, to update a segment of the queryString for the component in question.
- Updated the `ModelDetails` component so that it only shows a single table depending on the selected view from the aforementioned `ButtonGroup`.
- 

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- The application should log you in as expected.
- When viewing your models list you should be able to click the grouping buttons to change the grouping and the back button should work as expected.
- After clicking a model to view its details, clicking the view buttons should change the table that's shown and the back button should work as expected.
- Viewing the model details page at different widths, the view button group should move depending on the 3 breakpoints.

## Details

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1365

## Screenshots

Screen Shot 2020-09-10 at 10 02 12 PM

### CODE REVIEW

File: src/components/ButtonGroup/ButtonGroup.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import classNames from "classnames";
+
+import "./_button-group.scss";
+
+const ButtonGroup = ({ buttons, label, activeButton, setActiveButton }) => {
```
Comment (hatched) [inline src/components/ButtonGroup/ButtonGroup.js:6]: Once this lands I'll be proposing this to the Vanilla WG.

File: src/components/FilterTags/FilterTags.js [hunk 2]
```diff
@@ -109,14 +109,14 @@ const FilterTags = () => {
 
   // Update query params when adding and removing filters
   useEffect(() => {
-    const queryParams = queryString.parse(location.search);
+    const queryParams = queryString.parse(window.location.search);
```
Comment (hatched) [inline src/components/FilterTags/FilterTags.js:112]: This component needs to do a better job of it's queryString handling but I figured I should leave it until we switch to the new Filter UI.

File: src/components/LogIn/LogIn.js [hunk 3]
```diff
@@ -45,11 +45,10 @@ export default function LogIn({ children }) {
             {generateErrorMessage(loginError)}
           </div>
         </div>
-        <main>{children}</main>
```
Comment (hatched) [inline src/components/LogIn/LogIn.js:48]: This implementation was causing the whole app to be torn down and re-rendered after logging in.

File: src/pages/Models/Details/_model-details.scss [hunk 4]
```diff
@@ -26,10 +26,16 @@
...
+    display: grid;
+    grid-template-columns: 1fr;
 
     @media (min-width: $breakpoint-medium) {
-      display: flex;
+      gap: 1rem;
+      grid-template-columns: 240px 1fr;
+    }
+
+    @media (min-width: $breakpoint-m-large) {
```
Comment (hatched) [inline src/pages/Models/Details/_model-details.scss:37]: I'm not sure we want to be creating custom breakpoints, we should investigate instead overriding the vanilla `breakpoint-x-large` instead.

File: src/components/ButtonGroup/ButtonGroup.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <div className="p-button-group">
+      <div className="p-button-group__inner">
+        <span className="p-button-group__label">{label}</span>
+        <div className="p-button-group__buttons">
+          {buttons.map((label) => (
+            <button
+              aria-label={`view by ${label}`}
```
Comment (barrymcgee) [inline src/components/ButtonGroup/ButtonGroup.js:14]:
```suggestion
              aria-label={`View by ${label}`}
```

**Comment by barrymcgee** [comment]
Select several filter tags and refresh page:

Screenshot 2020-09-11 at 10 33 36

File: src/pages/Models/Details/_model-details.scss (hunk 4 above)
Comment (barrymcgee) [reply src/pages/Models/Details/_model-details.scss:37]: We don't want to override vanilla breakpoints that may break any components relying on that existing breakpoint value.

File: src/components/FilterTags/FilterTags.js (hunk 2 above)
Comment (barrymcgee) [reply src/components/FilterTags/FilterTags.js:112]: Shouldn't this use your new useQuery hook?

**Comment by hatched** [comment]
The demo service doesn't support loading the dashboard from anything other than `/`.

File: src/components/FilterTags/FilterTags.js (hunk 2 above)
Comment (hatched) [reply src/components/FilterTags/FilterTags.js:112]: Yes but this PR was already feature creeping considerably.

File: src/components/ButtonGroup/ButtonGroup.js (hunk 5 above)
Comment (barrymcgee) [reply src/components/ButtonGroup/ButtonGroup.js:14]: Screen readers are not the only assistive technology to use ARIA. However, some screen readers use capitalisation as cues for intonation.

Comment (hatched) [reply src/components/ButtonGroup/ButtonGroup.js:14]: Cool, I'll make a note to go update the rest of them in a follow-up.

**Review by barrymcgee — APPROVED** [review]

**Pull request merged**